### ISSUE TITLE
Choose name for PR

### ISSUE THREAD

**Issue by Ghostbird** [microsoft/vscode-pull-request-github#906]
I don't like one thing about #661: It defaults the PR title to the last commit message. However, whenever I make a PR, my last commit is generally not the main commit of the branch, but a clean-up or maintenance commit. Which means that my PRs are almost always created with nonsensical titles.

I suggest this workflow:

0. Open Command Palette
1. _Github Pull Requests: Create Pull Request_
2. Pop-up asks for branch
3. Enter branch
5. Press <kbd>Enter</kbd>
6. Pop-up asks for PR name
    * Pre-filled with last commit message
    * Pre-selected so starting to type immediately erases the old value.
7. [_Optional_] Type new PR name
8. Press <kbd>Enter</kbd>

To get the same results as the current workflow, the only change is that you have to double-tap <kbd>Enter</kbd> to immediately accept the suggested PR name. In the current workflow you would press <kbd>Enter</kbd> only once. I think this is an acceptable trade-off.

**Comment by GeorgeWL** [comment]
I'd also suggest to make this an opt-in in the settings however, as some people do the workflow of commit message titles

**Comment by JCQuintas** [comment]
I agree with @daedalus28, I would expect a github tool to behave just like github web interface does. The way it is implemented right now feels a bit clumsy, where you have to create the PR and then clean up the things it did that you didn't really want or need.

**Comment by phifa** [comment]
Maybe a setting, where you can choose where the suggested title is coming from, either from the branch name or last commit, or custom.

**Comment by GeorgeWL** [comment]
I don't see why it can't be a select box with:

```html
<select>
     <option>Use Latest Commit Message</option
     <option>Use Branch Name</option
     <option>Ask with Prompt Every Time</option> // default to if press esc, use last commit message
</select>
```

**Comment by Ghostbird** [comment]
Thanks so much for this feature!

**Comment by GeorgeWL** [comment]
Yeah this is honestly one of the best features added this year.

**Comment by GeorgeWL** [comment]
> More generally, I think just being able to open a draft of a PR where we can edit any of the properties (assignee, labels, etc) before actually creating it is probably the way to go. That would give it feature parity with github's web interface for creating PRs (which this is attempting to displace).

Draft PR's I feel should be a feature, but also should have their own Issue for it.

**Comment by RMacfarlane** [comment]
@GeorgeWL There should be an existing command for draft PRs, `GitHub Pull Requests: Create Draft Pull Request`. Perhaps it would be more discoverable if we had a single command that would ask if you wanted to publish the PR as a draft or not at the start?

**Comment by GeorgeWL** [comment]
I think what they're proposing however is just a setting in the settings ui for "Always Draft PRs" or similar.

@RMacfarlane 

I think single command would end up making the process slower for all those who don't have any interest in Draft PRs tbh.
